Tickets/dm 30366

bxin opened the pull request:
Notebook for
Slew simulation with M1M3 (cell+surrogate)

bxin committed:
4/22/21 M1M3 slew test

bxin committed:
improve comments

hdrass commented on bxin/ptg2m1m3/m1m3_diagnostic.ipynb:
```diff
@@ -6,8 +6,11 @@
    "source": [
```
<p>Defining the log message with the test case number and a descriptive name for the test would be good. </p><p>Also, I think it would be good to send a log message to EFD with the log message, a description of the test and the timestamp. This will make it easier later to go back.</p><p>Not sure, if this comes later in this notebook.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

bxin replied:
done. (thanks for your code snippet)

hdrass commented on bxin/ptg2m1m3/m1m3_diagnostic.ipynb:
```diff
@@ -6,8 +6,11 @@
    "source": [
```
<p>This are definition that are in the your library now, right? Maybe better to take this out and only keep the library up-to-date? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

hdrass replied:
<p>To the library? All def to the library? we will need them anyways in different contexts</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

hdrass replied:
<pre class="ql-syntax" spellcheck="false">Line #5.    #    ask Petr to do this.
</pre><p>I think here you now have a method on how to do it, right? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='64'/>

hdrass replied:
<p>same as before.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='77'/>

hdrass commented on bxin/ptg2m1m3/m1m3_diagnostic.ipynb:
```diff
@@ -6,8 +6,11 @@
    "source": [
```
<p>Let put all imports to the beginning and maybe to the library since we most of the time use the same imports.</p><p>Lets define the CSC index for the hexapod together with summit vs NCSA since this is pretty global for this notebook.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

bxin replied:
done

hdrass commented on bxin/ptg2m1m3/m1m3_diagnostic.ipynb:
```diff
@@ -6,8 +6,11 @@
    "source": [
```
<p>If I got Russel right, we should use aget only the first time and the .get. Leave this comment here for all the coming aget :-)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

bxin replied:
Tiago has told me never to use `.get` in a notebook. Based on slack chats between Russel, Tiago, and myself, I am following Tiago's recommendation

hdrass commented on bxin/ptg2m1m3/m1m3_diagnostic.ipynb:
```diff
@@ -6,8 +6,11 @@
    "source": [
```
<p>We do not use the controllers anymore, right? Maybe update here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/49/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='74'/>

bxin replied:
correct. we only used the controllers for initial communication tests. But I think let's keep it here because it may be helpful later for lower level debugging (where we don't want to involve the mount).

hdrass reviewed (approved):
Hi Bo,
please find my comments in your code.
In general, all looks good. There are a couple of changes that you might want to include to reflect the developments since the last revision of the notebook. These are are optimizations. Functionality should all be good, so I approve. It would be great, if we could see the notebook in action together.

bxin committed:
address comments from Holger

bxin commented:
@hdrass I addressed some of your comments. 
Just for the record - 
For those comments on using the latest methods, we agreed that there is no need to go back and update, because (1) these notebooks are in my personal area, and (2) they are only for preliminary level 3 tests.

Thanks again for the review. I am merging now...
